app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_posts_counter
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] OPTIONAL: Same here,  I would recommend to increment counter every time by 1,  to do that you can simply use : 

```
 def update_post_counter
  author.increment!(:posts_counter)
end
```